pkg/operator/hive/hive.go

```diff
@@ -211,3 +215,23 @@ func (r *ReconcileHiveConfig) includeAdditionalCAs(hLog log.FieldLogger, h *reso
 
 	return nil
 }
+
+func (r *ReconcileHiveConfig) includeGlobalPullSecret(hLog log.FieldLogger, h *resource.Helper, instance *hivev1.HiveConfig, hiveDeployment *appsv1.Deployment) error {
+	if instance.Spec.GlobalPullSecret == nil {
+		hLog.Debug("GlobalPullSecret is not provided in HiveConfig, it will not be deployed")
+		return nil
+	}
+	globalPullSecretEnvVars := []corev1.EnvVar{
+		{
+			Name: "GLOBAL_PULL_SECRET",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: *instance.Spec.GlobalPullSecret,
+					Key:                  corev1.DockerConfigJsonKey,
+				},
```

Comment:
What happens when the secret reference in the hive config changes? Does the hive controller deploy a new hive? The answer to that should inform the proper way of dealing with what happens when the referenced secret changes.

Reply:
If we are not re-deploying hive, then the clusterdeployment controller needs to be able to read the secret reference directly from the hiveconfig instead of getting anything passed to it via environment variables.

Reply:
HiveConfig is hive-operator's operand, which we used to generate all the other Deployments for things like hive-controller, we discussed and decided it was probably a bad idea to introduce a dependency to the operand into the code like that. At present hive-operator is the only component that reads from HiveConfig, and everything else is configured via env vars.

If we switch to env var = a secret name, and the secret ref in HiveConfig changes, the hive-operator will reconcile, adjust the env var in the hive-controllers Deployment, and thus kube should cycle the pod.

If the contents of the secret change, we pick that up the first time we use it because we're loading it on the fly in hive-controllers.

In it's current form I think the former is fine, but the latter is not.